src/pages/Blog.js

```diff
@@ -1,11 +1,182 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useContext } from "react";
 import Container from "@material-ui/core/Container";
+import { useParams } from "react-router-dom";
+import { loadBlogPost } from "../data/feedLibrary";
+import { SkynetContext } from "../state/SkynetContext";
+import ReactMarkdown from "react-markdown";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import CardHeader from "@material-ui/core/CardHeader";
+import Avatar from "@material-ui/core/Avatar";
+import IconButton from "@material-ui/core/IconButton";
+import Twitter from "@material-ui/icons/Twitter";
+import GitHub from "@material-ui/icons/GitHub";
+import Telegram from "@material-ui/icons/Telegram";
+import Skeleton from "@material-ui/lab/Skeleton";
+import { displayName } from "../data/displayName";
+import { useHistory } from "react-router-dom";
+import CardActionArea from "@material-ui/core/CardActionArea";
+import { makeStyles } from "@material-ui/core/styles";
+
+//style for removing hover shading
+const useStyles = makeStyles((theme) => ({
+  focus: {
+    "&&&": {
+      opacity: 0,
+    },
+  },
+}));
 
 //Blog page component, used to create view blog posts, returns JSX layout for rendering blogs
 export default function Blog(props) {
+  const { getUserProfile, feedDAC } = useContext(SkynetContext); //values to use from Skynet context
+  let { ref, dac, domain, posts, file, id } = useParams(); //get parameters from the route
+  const [postData, setPostData] = useState(); //state for blog post data
+  const [isLoading, setLoading] = useState(true); //state to render loading indicators
+  const [date, setDate] = useState(null); //display date
+  const [author, setAuthor] = useState(null); //author of the blog, userProfile DAC profile object
+  const history = useHistory(); //react router hook
+  const classes = useStyles(); //const to useStyles in JSX
+
+  //execute this effect on entry and when the feedDAC connection status is valid
+  useEffect(() => {
+    if (ref != null && feedDAC.connector) {
+      getPostData();
+    }
+  }, [ref, feedDAC.connector]);
+
+  //handle using parameters to assemle post data including author and blog post data
+  const getPostData = async () => {
+    const val =
```

Comment:
Follow up: this might be worth refactoring into its own function so that it can be unit tested.